cns/service.go

```diff
@@ -224,9 +237,50 @@ func getTLSConfigFromKeyVault(tlsSettings localtls.TlsSettings, errChan chan<- e
 		},
 	}
 
+	if tlsSettings.UseMTLS {
+		tlsCert := cr.GetCertificate()
+		rootCAs, err := mtlsRootCAsFromCertificate(tlsCert)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get root CAs for configuring mTLS")
+		}
+		tlsConfig.ClientAuth = tls.RequireAndVerifyClientCert
+		tlsConfig.ClientCAs = rootCAs
+		tlsConfig.RootCAs = rootCAs
+	}
+
+	log.Debugf("TLS configured successfully from KV: %+v", tlsSettings)
+
 	return &tlsConfig, nil
 }
 
+// Given a TLS cert, return the root CAs
+func mtlsRootCAsFromCertificate(tlsCert *tls.Certificate) (*x509.CertPool, error) {
+	certPool := x509.NewCertPool()
+
+	// self signed, we assume this is a root CA
+	if len(tlsCert.Certificate) == 1 {
```

Comment:
if tlsCert length is not 1, what would be the case and should we continue to do certPool.AddCert on line 278?

Reply:
it'll break - i can add guards to make this more robust

Reply:
refactored to add guard clauses and added unit tests